### ISSUE TITLE
test_up_down_consistency broken after updating to pytest 6.2.4

### ISSUE THREAD

**Issue by cachance7** [schireson/pytest-alembic#23]
I updated pytest from 6.2.3 -> 6.2.4 today and `test_up_down_consistency` now fails when I try to run it. Below is the output from running with 6.2.3 vs 6.2.4. Is there anything else I can provide that would help debugging? Great plugin btw.

##### Success run w/ pytest 6.2.3
```bash
$ pipenv run pytest -vvv -s --test-alembic -k "pytest_alembic::test_up_down_consistency"
======================================================================================== test session starts =========================================================================================
platform darwin -- Python 3.8.6, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/bin/python
cachedir: .pytest_cache
Using --randomly-seed=2967129046
rootdir: <omitted>, configfile: pytest.ini, testpaths: tests
plugins: randomly-3.7.0, alembic-0.2.6, cov-2.11.1, snapshottest-0.6.0, mock-resources-1.5.1, timeout-1.4.2, flask-1.2.0
collected 388 items / 391 deselected

tests::pytest_alembic::test_up_down_consistency <- . running migrations
PASSED

========================================================================================== warnings summary ==========================================================================================
../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/aiodataloader.py:2
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/aiodataloader.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Iterable, namedtuple

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/graphene/types/field.py:2
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/graphene/types/field.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Mapping, OrderedDict

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/newrelic/console.py:84: 18 warnings
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/newrelic/console.py:84: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
    prototype = wrapper.__name__[3:] + ' ' + inspect.formatargspec(

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/_pytest/nodes.py:274
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/_pytest/nodes.py:274: PytestUnknownMarkWarning: Unknown pytest.mark.mysql - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    marker_ = getattr(MARK_GEN, marker)

tests::pytest_alembic::test_up_down_consistency
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/pytest_alembic/plugin/plugin.py:98: PytestDeprecationWarning: A private pytest class or function was used.
    fixture_request = fixtures.FixtureRequest(self)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================== 1 passed, 391 deselected, 22 warnings in 21.22s ===========================================================================
```

##### Failed run w/ pytest 6.2.4
```bash
$ pipenv run pytest -vvv -s --test-alembic -k "pytest_alembic::test_up_down_consistency"
======================================================================================== test session starts =========================================================================================
platform darwin -- Python 3.8.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/bin/python
cachedir: .pytest_cache
Using --randomly-seed=1002127626
rootdir: <omitted>, configfile: pytest.ini, testpaths: tests
plugins: randomly-3.7.0, alembic-0.2.6, cov-2.11.1, snapshottest-0.6.0, mock-resources-1.5.1, timeout-1.4.2, flask-1.2.0
collected 388 items / 391 deselected

tests::pytest_alembic::test_up_down_consistency <- . running migrations
FAILED

============================================================================================== FAILURES ==============================================================================================
__________________________________________________________________ [pytest-alembic] tests::pytest_alembic::test_up_down_consistency __________________________________________________________________
Failing Revision:
    head

Alembic Error:
    Not a valid downgrade target from current heads

Errors:
E   pytest_alembic.plugin.error.AlembicTestFailure: Failed to downgrade through each revision individually.
========================================================================================== warnings summary ==========================================================================================
../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/aiodataloader.py:2
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/aiodataloader.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Iterable, namedtuple

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/graphene/types/field.py:2
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/graphene/types/field.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Mapping, OrderedDict

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/newrelic/console.py:84: 18 warnings
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/newrelic/console.py:84: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
    prototype = wrapper.__name__[3:] + ' ' + inspect.formatargspec(

../../../.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/_pytest/nodes.py:274
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/_pytest/nodes.py:274: PytestUnknownMarkWarning: Unknown pytest.mark.mysql - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    marker_ = getattr(MARK_GEN, marker)

tests::pytest_alembic::test_up_down_consistency
  /Users/caseychance/.local/share/virtualenvs/website-Mh3F1bnu/lib/python3.8/site-packages/pytest_alembic/plugin/plugin.py:98: PytestDeprecationWarning: A private pytest class or function was used.
    fixture_request = fixtures.FixtureRequest(self)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================================================================================== short test summary info =======================================================================================
FAILED tests::pytest_alembic::test_up_down_consistency
========================================================================== 1 failed, 391 deselected, 22 warnings in 20.25s ===========================================================================
make: *** [test-alembic] Error 1

```

**Comment by DanCardin** [comment]
Thanks for using the plugin!

I wonder if this is actually related to https://github.com/schireson/pytest-alembic/issues/19, i.e. how we before assumed (for all built-in tests, even if one disabled the `single_head_revision` test) single head revisions. Based on the alembic-produced "Not a valid downgrade target from current heads" message, it **seems** like you have more than one head revision. If you can confirm that, I might have incidentally fixed it with 0.3.0.

The concerning part (if 0.3.0 doesn't fix for example), is why your tests are failing if the pytest version is the **only** change on your end. i manually installed 6.3.4 for pytest-alembic's tests and haven't managed to get it to fail in any similar way. Ideally if you could produce a minimally failing alembic history, i can easily turn it into an example/testcase see what's going on.

**Comment by cachance7** [comment]
There have been no branches in the history and only one head which points to the latest revision. 

After upgrading to pytest 6.2.4 & pytest-alembic 0.3.0 I'm happy to report that I no longer experience the issue. Unfortunately, I'm also now unable to downgrade and offer a minimal repro using pytest 6.2.4 & alembic 0.2.6 – reverting to the originally broken git commit and pipenv state does not yield a broken test now. 

Just going to close this now since I can't offer anything more. Perhaps I overlooked something at the time and the pytest version wasn't the only thing to change as you suggested. Appreciate the response and you investigating!
